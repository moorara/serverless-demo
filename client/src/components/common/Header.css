.app-logo {
  animation: app-logo-spin infinite 20s linear;
  height: 80px;
}

@keyframes app-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

a.is-active {
  text-transform: uppercase;
}
