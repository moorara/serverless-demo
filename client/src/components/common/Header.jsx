import './Header.css'

import React from 'react'
import PropTypes from 'prop-types'
import { Route, NavLink } from 'react-router-dom'

import logo from './logo.png'
import Loading from './Loading'

const TabItem = ({ to, children }) => (
  <Route exact path={to} children={
    ({ location, match }) => (
      <li className={match ? "is-active" : ""}>
        <NavLink to={to} exact activeClassName="is-active">{children}</NavLink>
      </li>
    )
  } />
)

TabItem.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.string.isRequired
}

const Header = ({ isLoading }) => (
  <section className="hero is-primary is-medium is-bold">
    <div className="hero-head">
      <div className="container">
        <div className="columns">
          <div className="column is-2 is-offset-5">
            { isLoading && <Loading dots={20} interval={100} /> }
          </div>
        </div>
      </div>
    </div>

    <div className="hero-body">
      <div className="container">
        <div className="columns">
          <div className="column is-1">
            <img src={logo} className="app-logo" alt="logo" />
          </div>
          <div className="column">
            <h1 className="title">Serverless Application</h1>
            <h2 className="subtitle">A React client for consuming serverless API!</h2>
          </div>
        </div>
      </div>
    </div>

    <div className="hero-foot">
      <nav className="tabs is-boxed">
        <div className="container">
          <ul>
            <TabItem to="/">Home</TabItem>
            <TabItem to="/about">About</TabItem>
          </ul>
        </div>        
      </nav>
    </div>
  </section>
)

Header.propTypes = {
  isLoading: PropTypes.bool.isRequired
}

export default Header
