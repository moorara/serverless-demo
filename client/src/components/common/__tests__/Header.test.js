import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'

import Header from '../Header'

describe('Header', () => {
  test('renders react component with isLoading false', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Header isLoading={false} />
      </MemoryRouter>
    )
    const tree = component.toJSON()
    const instance = component.root

    expect(tree).toMatchSnapshot()
  })
  test('renders react component with isLoading true', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Header isLoading={true} />
      </MemoryRouter>
    )
    const tree = component.toJSON()
    const instance = component.root

    expect(tree).toMatchSnapshot()
  })
})
