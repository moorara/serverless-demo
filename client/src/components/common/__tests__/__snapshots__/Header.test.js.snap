// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders react component with isLoading false 1`] = `
<section
  className="hero is-primary is-medium is-bold"
>
  <div
    className="hero-head"
  >
    <div
      className="container"
    >
      <div
        className="columns"
      >
        <div
          className="column is-2 is-offset-5"
        />
      </div>
    </div>
  </div>
  <div
    className="hero-body"
  >
    <div
      className="container"
    >
      <div
        className="columns"
      >
        <div
          className="column is-1"
        >
          <img
            alt="logo"
            className="app-logo"
            src="logo.png"
          />
        </div>
        <div
          className="column"
        >
          <h1
            className="title"
          >
            Serverless Application
          </h1>
          <h2
            className="subtitle"
          >
            A React client for consuming serverless API!
          </h2>
        </div>
      </div>
    </div>
  </div>
  <div
    className="hero-foot"
  >
    <nav
      className="tabs is-boxed"
    >
      <div
        className="container"
      >
        <ul>
          <li
            className="is-active"
          >
            <a
              aria-current="page"
              className="is-active"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              Home
            </a>
          </li>
          <li
            className=""
          >
            <a
              aria-current={null}
              href="/about"
              onClick={[Function]}
            >
              About
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</section>
`;

exports[`Header renders react component with isLoading true 1`] = `
<section
  className="hero is-primary is-medium is-bold"
>
  <div
    className="hero-head"
  >
    <div
      className="container"
    >
      <div
        className="columns"
      >
        <div
          className="column is-2 is-offset-5"
        >
          <div
            className="dots"
          >
            <span>
              ••
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="hero-body"
  >
    <div
      className="container"
    >
      <div
        className="columns"
      >
        <div
          className="column is-1"
        >
          <img
            alt="logo"
            className="app-logo"
            src="logo.png"
          />
        </div>
        <div
          className="column"
        >
          <h1
            className="title"
          >
            Serverless Application
          </h1>
          <h2
            className="subtitle"
          >
            A React client for consuming serverless API!
          </h2>
        </div>
      </div>
    </div>
  </div>
  <div
    className="hero-foot"
  >
    <nav
      className="tabs is-boxed"
    >
      <div
        className="container"
      >
        <ul>
          <li
            className="is-active"
          >
            <a
              aria-current="page"
              className="is-active"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              Home
            </a>
          </li>
          <li
            className=""
          >
            <a
              aria-current={null}
              href="/about"
              onClick={[Function]}
            >
              About
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</section>
`;
